modin/logging/logger_base_class.py

```diff
@@ -0,0 +1,56 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""
+Module contains ``LoggerBase`` class.
+
+``LoggerBase`` is used for adding logging to Modin classes and their subclasses.
+"""
+from typing import Optional
+
+from .logger_function import logger_decorator
+
+
+class LoggerBase:
```

Comment:
`Logger` is finer to me 😄

Reply:
We have `logger_decorator.py`, so would prefer this name to not be just `logger.py`, but rather a bit more descriptive. What about `ClassLogger`?

Reply:
Maybe it's better be `LoggerClass` then? I'm no native speaker, but `ClassLogger` sounds like it's logging some classes...

Reply:
Don't both `class ClassLogger` and `class LoggerClass` sound like a bit duplicated? I still tend to stick to just `Logger`.

Reply:
cc @vnlitvinov, @naren-ponder

Reply:
Outside of the definition, we would be doing `from modin.logging import ClassLogger` and then attaching it to different classes so that naming convention makes sense to me. If you have a strong objection, I am okay changing to `Logger`.

Reply:
I don't like using `Class` word in class names or `function` word in function names because it looks like a repetition. I just want to be sure that we won't want to rename it in future. If @vnlitvinov is okay to keep just `Logger` too, let's use this name.

Reply:
It's not exactly a repetition, and I feel that just `Logger` would introduce a confusion as it would overlap with `logging.Logger` (from the standard library) _but_ it has completely different functionality.

I would rather leave it be a `ClassLogger` or invent a better name describing its functionality (like maybe `LogEnabler`?..), but I'm certainly against `Logger` here.